packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -454,6 +454,30 @@ export class DataStores implements IDisposable {
         return builder.getGCData();
     }
 
+    /**
+     * After GC has run, called to notify this Container's data stores of routes that are used in it.
+     * @param usedRoutes - The routes that are used in all data stores in this Container.
+     */
+    public updateUsedRoutes(usedRoutes: string[]) {
+        // Build a map of data store ids to routes used in it.
+        const usedRoutesMap: Map<string, string[]> = new Map();
+        for (const route of usedRoutes) {
+            const dataStoreId = route.split("/")[1];
+            const dataStoreRoute = route.slice(dataStoreId.length + 1);
+            const routes = usedRoutesMap.get(dataStoreId);
+            if (routes !== undefined) {
+                routes.push(dataStoreRoute);
+            } else {
+                usedRoutesMap.set(dataStoreId, [dataStoreRoute]);
+            }
+        }
+
+        // Update the used routes in each data store. Used routes is empty for unused data stores.
+        for (const [contextId, context] of this.contexts) {
```

Comment:
Also may want to consider the case of routes that do not have a matching entry in `this.contexts`.  Not sure what to do in those cases: maybe either assert, add telemetry, or leave a comment here explaining the meaning of that case.